src/main/java/com/servlet/repository/StudentRepository.java

```diff
@@ -0,0 +1,129 @@
+package com.servlet.repository;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import com.servlet.entity.Student;
+import com.servlet.provider.SQLConnectionProvider;
+
+public class StudentRepository {
+
+    private final SQLConnectionProvider connectionProvider;
+
+    public StudentRepository(SQLConnectionProvider connectionProvider) {
+        this.connectionProvider = connectionProvider;
+    }
+
+    public List<Student> all() {
+        List<Student> students = new ArrayList<Student>();
+
+        try {
+            Connection conn = connectionProvider.getConnection();
```

Comment:
Delete this and create field Connection conn = connectionProvider.getConnection(); outside your methods and than use it in them.